<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Kaapi: Exception</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.6.3 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>Exception<br/>
<small>
[<a class="el" href="group___util.html">Util</a>]</small>
</h1>
<p>Definition of exceptions in KAAPI.  
<a href="#_details">More...</a></p>

<p><div class="dynheader">
Collaboration diagram for Exception:</div>
<div class="dynsection">
<center><table><tr><td><img src="group___exception.png" border="0" alt="" usemap="#group______exception_map"/>
<map name="group______exception_map" id="group______exception">
<area shape="rect" id="node1" href="group___util.html" title="Basic interface for operating system specification." alt="" coords="5,5,51,35"/></map></td></tr></table></center>
</div>
</p>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Classes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_exception.html">Util::Exception</a></td></tr>
<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Base class for the KAAPI's exception hierarchy Base class for hierarchy of error classes. Match the ANSI/C++ proposal for the hierarchy.  <a href="class_util_1_1_exception.html#_details">More...</a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_restart_exception.html">Util::RestartException</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_server_exception.html">Util::ServerException</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_runtime_error.html">Util::RuntimeError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_posix_error.html">Util::PosixError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_range_error.html">Util::RangeError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_logic_error.html">Util::LogicError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_invalid_argument_error.html">Util::InvalidArgumentError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_length_error.html">Util::LengthError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_out_of_range_error.html">Util::OutOfRangeError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_no_more_resource.html">Util::NoMoreResource</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_bad_alloc.html">Util::BadAlloc</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_i_o_error.html">Util::IOError</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_end_of_file.html">Util::EndOfFile</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_com_failure.html">Util::ComFailure</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_bad_u_r_l.html">Util::BadURL</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_already_bind.html">Util::AlreadyBind</a></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">class &nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_no_found.html">Util::NoFound</a></td></tr>
<tr><td colspan="2"><h2>Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="group___exception.html#gabdddb1a0fee9d2ab4479037a4ed05dc4">Util::__Exception_throw</a> (const Exception &amp;err)</td></tr>
<tr><td class="memTemplParams" colspan="2">template&lt;class T &gt; </td></tr>
<tr><td class="memTemplItemLeft" align="right" valign="top">void&nbsp;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="group___exception.html#gaced456060748084f46d394f9cdb28dff">Util::Exception_throw</a> (const T &amp;ex)  throw (T)</td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>Definition of exceptions in KAAPI. </p>
<hr/><h2>Function Documentation</h2>
<a class="anchor" id="gabdddb1a0fee9d2ab4479037a4ed05dc4"></a><!-- doxytag: member="Util::__Exception_throw" ref="gabdddb1a0fee9d2ab4479037a4ed05dc4" args="(const Exception &amp;err)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void Util::__Exception_throw </td>
          <td>(</td>
          <td class="paramtype">const Exception &amp;&nbsp;</td>
          <td class="paramname"> <em>err</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Function call by KAAPI_ASSERT before throwing a Kaapi exception The main goal of this function is to be able to capture and exception during debuging the program. All exception throwed by Kaapi make a call to this function. </p>
<dl class="see"><dt><b>See also:</b></dt><dd><a class="el" href="group___util.html#ga78fac47cf1e021aca7ee804ad7b950e3" title="Assertion which display message before calling System_abort if cond is failed.">KAAPI_ASSERT</a> </dd></dl>

</div>
</div>
<a class="anchor" id="gaced456060748084f46d394f9cdb28dff"></a><!-- doxytag: member="Util::Exception_throw" ref="gaced456060748084f46d394f9cdb28dff" args="(const T &amp;ex)" -->
<div class="memitem">
<div class="memproto">
<div class="memtemplate">
template&lt;class T &gt; </div>
      <table class="memname">
        <tr>
          <td class="memname">void Util::Exception_throw </td>
          <td>(</td>
          <td class="paramtype">const T &amp;&nbsp;</td>
          <td class="paramname"> <em>ex</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td>  throw (T)<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Function call by macro to throw a Kaapi exception </p>
<dl class="see"><dt><b>See also:</b></dt><dd><a class="el" href="group___util.html#ga78fac47cf1e021aca7ee804ad7b950e3" title="Assertion which display message before calling System_abort if cond is failed.">KAAPI_ASSERT</a></dd></dl>
<p>Function to throw an exception. The purpose of this method is to be able to set a breakpoint on throw </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>ex</em>&nbsp;</td><td>a reference to an <a class="el" href="class_util_1_1_exception.html" title="Base class for the KAAPI&#39;s exception hierarchy Base class for hierarchy of error...">Exception</a> object </td></tr>
  </table>
  </dd>
</dl>
<dl><dt><b>Exceptions:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>the</em>&nbsp;</td><td>err object </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
</div>
<hr class="footer"/><address style="text-align: right;"><small>Generated on Wed Jun 16 08:51:57 2010 for Kaapi by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.3 </small></address>
</body>
</html>
